dts/bindings/pinctrl/st,stm32-pinmux.yaml

```diff
@@ -11,6 +11,50 @@ properties:
     reg:
       required: true
 
-pinmux-cells:
-  - pin
-  - function
+child-binding:
```

Comment:
In my view the usage of child-binding here is not optimal. But as you are following  #20386 it is clear to do so. You may have a look at #22255 which is the solution we are using for the OOT pinctrl driver.